src/main/java/LogicManager.java

```diff
@@ -0,0 +1,172 @@
+        MarkUnmarkArguments markUnmarkArguments = new MarkUnmarkArguments(taskIndex);
+        if (words[0].equals("mark")) {
+            return new Command(CommandType.MARK, markUnmarkArguments);
+        }
+        return new Command(CommandType.UNMARK, markUnmarkArguments);
+    }
+
+    private Command getToDoCommand(String currentInput) {
+        // Extract after _todo_, which is 4 characters
+        String taskName = currentInput.substring(4);
+        taskName = taskName.trim();
+        if (taskName.isEmpty()) {
+            return getInvalidCommand();
+        }
+        return new Command(CommandType.TODO, new ToDoArguments(taskName));
+    }
+    private Command getDeadlineCommand(String currentInput) {
+        try {
+            int idxOfDeadline = currentInput.indexOf("/");
+
+            // Extract after _deadline_, which is 8 characters long
+            String taskName = currentInput.substring(8, idxOfDeadline);
+            taskName = taskName.trim();
+            if (taskName.isEmpty()) {
+                return getInvalidCommand();
+            }
+
+            // We extract the deadline after _/by _, which is 4 characters long
+            String by = currentInput.substring(idxOfDeadline + 4);
+            by = by.trim();
+
+            DeadLineArguments deadLineArguments = new DeadLineArguments(taskName, by);
+            return new Command(CommandType.DEADLINE, deadLineArguments);
+        } catch (Exception exception) {
+            return getInvalidCommand();
+        }
+    }
+    private Command getEventCommand(String currentInput) {
+        try {
+            int idxOfStart = currentInput.indexOf("/");
+            // Extract after _event_, which is 5 characters long
+            String taskName = currentInput.substring(5, idxOfStart);
+            taskName = taskName.trim();
+            if (taskName.isEmpty()) {
+                return getInvalidCommand();
+            }
+
+            int idxOfEnd = currentInput.indexOf("/", idxOfStart + 1);
+            // Extract start after _/from_, which is 5 characters long, and before idxOfEnd
+            String start = currentInput.substring(idxOfStart + 5, idxOfEnd);
+            start = start.trim();
+
+            // Extract end after _/to_, which is 3 characters long
+            String end = currentInput.substring(idxOfEnd + 3);
+            end = end.trim();
+
+            EventArguments eventArguments = new EventArguments(taskName, start, end);
+            return new Command(CommandType.EVENT, eventArguments);
+        } catch (Exception exception) {
+            return getInvalidCommand();
+        }
+    }
+
+    private Command getCommand(String currentInput) {
+        String[] words = currentInput.split(" ");
+        String commandType = words[0];
+        if (commandType.equals("list")) {
+            return getListCommand(words);
+        }
+        if (commandType.equals("mark") || commandType.equals("unmark")) {
```

Comment:
```suggestion
        else if (commandType.equals("mark") || commandType.equals("unmark")) {
```

Perhaps you should use else if. For the current Levels, since there is only one command, using else if would be more efficient since it won't iterate through all the ifs statements